src/sql/store.py

```diff
@@ -116,8 +120,17 @@ def __str__(self) -> str:
         template = (
             with_clause_template_backtick if is_use_backtick else with_clause_template
         )
+        # Remove trailing semicolon from query
+        query_without_trailing_semicolon = self._query
+        if self._query.strip().endswith(";"):
+            query_without_trailing_semicolon = self._query[::-1].replace(";", "", 1)[
```

Comment:
can youu explain this line? why do you need to reverse the string?

I think if the condition holds true, stripping and removing the last character should do the trick

Reply:
i did it this way because i wanted to avoid removing any newlines or spaces, just was trying to be least invasive to any formatting

Reply:
> i did it this way because i wanted to avoid removing any newlines or spaces, just was trying to be least invasive to any formatting

You can also add the test case cover this case, to prove the idea of this section

Reply:
> i did it this way because i wanted to avoid removing any newlines or spaces, just was trying to be least invasive to any formatting

I understand now! please just add some comment to explain the logic